src/odemis/acq/fastem.py

```diff
@@ -875,7 +875,13 @@ def correct_beam_shift(self):
 
         cur_beam_shift_pos = numpy.array(self._beamshift.shift.value)
         logging.debug("Current beam shift: {} [m]".format(self._beamshift.shift.value))
-        self._beamshift.shift.value = (cur_beam_shift_pos + beam_shift_cor)
+        # TODO remove the below temporary fix when a proper solution is found
```

Comment:
the acquisition will also fail when there is no pattern, then on line 846 an error will be raised. Could you also add a catch for that?

Reply:
Should the try statement be on line 677 then? instead of two inside the function

Reply:
I think it is good to know the specific case for which it failed, so you can keep it inside the function. Or try to catch both independently on line 677.